tools/codeowners-utils/Azure.Sdk.Tools.CodeownersUtils/Parsing/CodeownersEntry.cs

```diff
@@ -0,0 +1,171 @@
+    /// <code>
+    ///   # PRLabel: %Label1 %Label2
+    ///   # ServiceLabel: %Label3 %Label4
+    ///   path @owner @owner
+    /// </code>
+    /// <para>
+    /// ServiceLabel and ServiceOwners entry OR /&lt;NotInRepo&gt;/ (one or the other, not both). ServiceOwners is preferred going forward.
+    /// </para> 
+    /// <code>
+    ///   # ServiceLabel: %Label3 %Label4
+    ///   # ServiceOwners: @owner @owner
+    ///   or, the old style
+    ///   # ServiceLabel: %Label3 %Label4
+    ///   # /&lt;NotInRepo&gt;/: @owner @owner
+    /// </code>
+    /// </summary>
+    public class CodeownersEntry
+    {
+        public string PathExpression { get; set; } = "";
+
+        public bool ContainsWildcard => PathExpression.Contains('*');
+
+        public List<string> SourceOwners { get; set; } = new List<string>();
+
+        // PRLabels are tied to a source path/owner line
+        // In theory, there should only have be one PR Label.
+        public List<string> PRLabels { get; set; } = new List<string>();
+
+        // ServiceLabels are tied to either, ServiceOwners or /<NotInRepo>/ (MissingLabel moniker), or
+        // the owners from a source path/owners line (meaning that the owners for the source are also
+        // the service owners)
+        // In theory, there should only have ever be one ServiceLabel, aside from the ServiceAttention label
+        public List<string> ServiceLabels { get; set; } = new List<string>();
+
+        public List<string> ServiceOwners { get; set; } = new List<string>();
+
+        // AzureSdkOwners are directly tied to the source path. If the AzureSdkOwners are defined
+        // on the same line as the moniker, it'll use those, if it's empty it'll use the owners from
+        // the source path/owner line.
+        public List<string> AzureSdkOwners { get; set; } = new List<string>();
+
+        public bool IsValid => !string.IsNullOrWhiteSpace(PathExpression);
+
+        public CodeownersEntry()
+        {
+        }
+        public override string ToString()
+        {
+            return string.Join(
+                        Environment.NewLine,
+                        $"PathExpression:{PathExpression}, HasWildcard:{ContainsWildcard}",
+                        $"SourceOwners:{string.Join(", ", SourceOwners)}",
+                        $"PRLabels:{string.Join(", ", PRLabels)}",
+                        $"ServiceLabels:{string.Join(", ", ServiceLabels)}",
+                        $"ServiceOwners:{string.Join(", ", ServiceOwners)}",
+                        $"AzureSdkOwners:{string.Join(", ", AzureSdkOwners)}"
+                   );
+        }
+
+        /// <summary>
+        /// Remove all code owners which are not github alias.
+        /// Even with team expansion there can still be teams in lists. This
+        /// can happen if the team is not a child team under azure-sdk-write, which
+        /// are the only teams expanded.
+        /// </summary>
+        public void ExcludeNonUserAliases()
```

Comment:
We absolutely use this but not in the parser or linter. Both [notification-configuration](https://github.com/Azure/azure-sdk-tools/blob/main/tools/notification-configuration/notification-creator/Contacts.cs#L125) and [pipeline-owners-extractor](https://github.com/Azure/azure-sdk-tools/blob/main/tools/pipeline-owners-extractor/Azure.Sdk.Tools.PipelineOwnersExtractor/Processor.cs#L124) use this. It's still necessary to have this in there. Teams are expanded into users if and only if they're child teams of azure-sdk-write, otherwise the team is returned as part of the list of owners.